# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/MahamFatima5/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

## 1) My lane and why

**Provisional lane:** Refresh / Content Opportunity Scoring

**Why:** I already ran the starter notebooks, which use this same idea: a simple baseline rule is compared with a model to rank pages for review. It fits my next 7 weeks because the starter data and pipeline already work this way, and I can later test stronger labels on the full warehouse. I can change this lane until the end of Week 4.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

## 2) The question, decision, action, and cost of a wrong call

**Search question:** Which pages should be reviewed first because they are losing search visibility while still having enough demand to matter?

**Unit of analysis:** One page (one content item).

**Output:** A ranked list of pages to review, with a score and a short reason for each.

**Decision it improves:** Which pages a reviewer should spend limited time on first.

**Action someone could take:** Review the top 20 to 50 pages and decide whether to refresh them, monitor them, or leave them alone.

**Cost of a wrong recommendation:** Reviewer time is wasted on healthy pages, or a page that keeps declining is missed and loses more traffic.

**Why data or ML can help:** A fixed rule treats every page the same way. Data lets us check whether a learned ranking finds more truly declining pages in its top results than a simple rule does.

**Why this is not just "train a model":** The goal is a ranked review list for humans, checked against a simple baseline. The model has to beat the rule to be worth using, and it only supports decisions. It does not prove that a refresh fixes a page.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [1]:
import pandas as pd

url = "https://raw.githubusercontent.com/MahamFatima5/flyrank-ml-internship/main/data/raw/content_refresh_anonymized.csv"
df = pd.read_csv(url)

# Keep pages that have impressions and are at least 90 days old, one row per page
d = df[(df["impressions_90d"] > 0) & (df["content_age_days"] >= 90)]
d = d.drop_duplicates("content_id")

print("Pages in the starter file:", len(df))
print("Pages kept after filters:", len(d))
print("Share of pages trending down:", round((d["trend_direction"] == "down").mean() * 100, 1), "%")
print("Median impressions (90 days):", d["impressions_90d"].median())
print("Median content age (days):", d["content_age_days"].median())

Pages in the starter file: 30000
Pages kept after filters: 30000
Share of pages trending down: 54.2 %
Median impressions (90 days): 731.0
Median content age (days): 236.0


**What the numbers show:** After keeping pages with impressions and at least 90 days of age, 30,000 pages remain. About 54.2% of them are trending down, and the median page has about 731 impressions in 90 days. This shows enough pages and enough decline to make a review ranking worthwhile.

## 4) Careful words: what I can and cannot claim

**What I can say:**
- In this starter data, some pages are trending down while still having search demand.
- A ranked review list can help humans choose which pages to look at first.

**What I cannot say:**
- That a refresh will cause a page to recover.
- That I found a Google ranking factor.
- That these results hold for the full warehouse. This is a 30,000-row anonymized starter slice.

**Limits:** "Trending down" is a beginner proxy label based on the current window, not a future outcome. I will test stronger future-window labels later.

### Self-check

Before you submit, confirm each line honestly:

- ✅ Every section above is filled: markdown thinking AND the code that backs it
- ✅ The notebook runs top to bottom with no errors (Runtime → Run all)
- ✅ No client names, URLs, or private queries anywhere
- ✅ My claims use careful words: observed, measured, directional, decision-support
- ✅ Committed to my repo under `work/notebooks/` — then submit my repo URL on the card. Done.